Add auth_timeout parameter when supported by paramiko

orgito opened the pull request:
Paramiko 2.2 introduces the auth_timeout parameter. This will set the
parameter to the same value of the timeout parameter to prevent
"Authentication timeout" errors.

##### SUMMARY
If the paramiko version installed is at least 2.2 include the auth_timeout parameter to the connect method with the same value used for the timeout method. This will prevent "Authentication timeout" errors when a slow authentication step (>30s) happens with a host. This is common with some network devices.

Fixes #42596

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
paramiko_ssh connection plugin 

##### ADDITIONAL INFORMATION
I noticed that problem when using the network_cli connection plugin, which in turn uses the paramiko_ssh connection plugin.

orgito committed:
Add auth_timeout parameter when supported

Paramiko 2.2 introduces the auth_timeout parameter. This will set the
parameter to the same value of the timeout parameter to prevent
"Authentication timeout" errors.

sivel commented on lib/ansible/plugins/connection/paramiko_ssh.py:
```diff
@@ -337,17 +338,32 @@ def _connect_uncached(self):
+                    port=port,
+                    **sock_kwarg
+                )
+            else:
```
I think instead of having `ssh.connect` in an if/else, you should instead rename `sock_kwarg` to something like `kwargs` and conditionally add `auth_timeout` to `kwargs`

orgito replied:
Agree. Will do.

orgito replied:
@sivel done.

orgito committed:
Conditionally add auth_timeout to ssh.connect

Renamed sock_kwarg to ssh_connect_kwargs and conditionally added the
auth_timeout parameter based on the installed paramiko version.

dagwieers commented:
I am wondering whether there would ever be a need to set a different auth_timeout in paramiko ?
Other transports have separate values for connection_timeout/connect_timeout, read_timeout or operation_timeout, even when they default to another timeout value if unset.

orgito commented:
The advantage of using the same value is that it retains current expected behavior when using a version of paramiko older than 2.2. So I don't think it is necessary to handle an extra parameter here.

dagwieers reviewed (approved):
LGTM

samdoran commented:
Please create a changelog fragment. See [this fragment](https://github.com/ansible/ansible/blob/stable-2.7/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

orgito committed:
Add changelog fragment

orgito commented:
> Please create a changelog fragment. See [this fragment](https://github.com/ansible/ansible/blob/stable-2.7/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

Thank you. It is done

dagwieers commented:
rebuild_merge
